providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/steps/StepDefinitions.java

```diff
@@ -433,47 +433,53 @@ public void a_provider_ready_handler_is_added() {
     @Then("the PROVIDER_READY handler must run")
     public void the_provider_ready_handler_must_run() {
         Awaitility.await()
-        .atMost(Duration.ofSeconds(2))
-        .until(() -> {
-            return this.isReadyHandlerRun;
-        });
+                .atMost(Duration.ofSeconds(2))
+                .until(() -> {
+                    return this.isReadyHandlerRun;
+                });
     }
 
     /*
-    * Zero Value
-    */
+     * Zero Value
+     */
 
     // boolean value
     @When("a zero-value boolean flag with key {string} is evaluated with default value {string}")
-    public void a_zero_value_boolean_flag_with_key_is_evaluated_with_default_value(String flagKey, String defaultValue) {
+    public void a_zero_value_boolean_flag_with_key_is_evaluated_with_default_value(String flagKey,
+            String defaultValue) {
         this.booleanFlagKey = flagKey;
         this.booleanFlagDefaultValue = Boolean.valueOf(defaultValue);
     }
+
     @Then("the resolved boolean zero-value should be {string}")
     public void the_resolved_boolean_zero_value_should_be(String expected) {
         boolean value = client.getBooleanValue(this.booleanFlagKey, this.booleanFlagDefaultValue);
-         assertEquals(Boolean.valueOf(expected),value);   
+        assertEquals(Boolean.valueOf(expected), value);
     }
 
     // float/double value
     @When("a zero-value float flag with key {string} is evaluated with default value {double}")
     public void a_zero_value_float_flag_with_key_is_evaluated_with_default_value(String flagKey, Double defaultValue) {
```

Comment:
When testing locally defaultValue will not produce 0.0. Instead defaultValue is stuck at 0. Only the in-process evaluator seems to be impacted.

Reply:
@Kavindu-Dodan @beeme1mr  I think there's a bug here. I did a quick review and it seems @DBlanchard88 is right. I think there's a  interesting edge case where when one variant is `0.0` as is the case here, our in-process evaluator parses the variant values as `Integer`. I'm not sure if it matters if the `0.0` value flag comes first or not. See this screenshot from my debugger: 
![image](https://github.com/open-feature/java-sdk-contrib/assets/25272906/7dc11864-a8b1-480c-8359-53f6535dbf69)

I assume we can somehow configure the Json mapping annotations to help with this, but it might be tricky.

Reply:
@Kavindu-Dodan I'm going to open a bug for this and assign myself since I think I understand what's happening. Feel free to tell me I'm wrong if you think I've missed something.

Reply:
Yeah, this is intersting 🤔 I cannot reproduce this observation locally 

```json
    "float-zero-flag": {
      "state": "ENABLED",
      "variants": {
        "zero": 0.0,
        "non-zero": 1.0
      },
      "defaultVariant": "zero"
    }
```

And ,

![image](https://github.com/open-feature/java-sdk-contrib/assets/8186721/3824a453-50fa-4e14-8a2a-6a09eb4a0edd)


I have Java version `java 17.0.8 ` if that matters 🤔

Reply:
That's wild. I'm able to reproduce this on Java 8, 17, 20. As an experiment, and a sanity check, I changed the variant names (appended `"f"`) and set the non-zero value to `1.1`. When I do this, the non-zero variant is correctly a double, and the `0.0` variant is an int:

![image](https://github.com/open-feature/java-sdk-contrib/assets/25272906/abf07672-ad27-4909-a9a5-3b3ccc723d1e)

![image](https://github.com/open-feature/java-sdk-contrib/assets/25272906/dc58425c-57c3-4d46-82c8-61578bef2fc9)

Reply:
Yup, this fails locally and only when I am using grpc sync stream. THe example of mine above was a simple parsing from a file (sorry for not stating this)

So the loss of decimals happens either in wire OR at JSON decoding time. I will check in this direction

Reply:
The issue boils down to how to how Go unmarshel json to structs. While it store float64 values, decmials with trailing zeros are stored as integers (`0.0` -> `0`) 

This is why the sync stream sends us a `0` value and the evaluation fails. I think the best solution here is to handle such corner cases at the evaluator's end. 

In Java, we need special handling for Double - check if the variant can be converted to a double (`Double.parseDouble`) and use that value as the resolved value.

Reply:
> The issue boils down to how to how Go unmarshel json to structs. While it store float64 values, decmials with trailing zeros are stored as integers (`0.0` -> `0`)
> 
> This is why the sync stream sends us a `0` value and the evaluation fails. I think the best solution here is to handle such corner cases at the evaluator's end.
> 
> In Java, we need special handling for Double - check if the variant can be converted to a double (`Double.parseDouble`) and use that value as the resolved value.

Reply:
> The issue boils down to how to how Go unmarshel json to structs. While it store float64 values, decmials with trailing zeros are stored as integers (0.0 -> 0)

I was starting to suspect this was the case.

> In Java, we need special handling for Double - check if the variant can be converted to a double (Double.parseDouble) and use that value as the resolved value.

I was thinking this too... in fact, to be consistent with RPC, I think we have to attempt conversion in both directions, int -> float and float -> int. Additionally, if we do this, we may need to specify how the float -> int conversion is done... I think Java and Go both  round toward zero... I believe .NET does too. This might be worth specifying.